superset/viz.py

```diff
@@ -1263,17 +1263,17 @@ def to_series(
                     series_title = series_title + (title_suffix,)
 
             values = []
-            non_nan_cnt = 0
+            non_value_cnt = 0
             for ds in df.index:
                 if ds in ys:
                     d = {"x": ds, "y": ys[ds]}
-                    if not np.isnan(ys[ds]):
-                        non_nan_cnt += 1
+                    if not np.isnan(ys[ds]) and ys[ds] != 0:
+                        non_value_cnt += 1
```

Comment:
Instead of +1, we can probably change `non_value_cnt` to a boolean `has_value` and skip the `np.isnan` check as long as `has_value == True`.

Reply:
yeah, i noticed this too and thought the code was pretty janky. I think i can clean it up even more, regardless of what the proper approach is for this PR